### DISCUSSION TITLE
Legend margin

### DISCUSSION THREAD

**Discussion by dgwyer** [observablehq/plot#1080] [General]
I've looked at the docs but can't seem to get the legend margin working. I've tried a few things including adding a legend object to the style object as you can see here:

https://observablehq.com/@dgwyer/untitled/2

Not sure what I'm doing wrong.

**Comment by mbostock** [comment]
Like so?

```js
  color: {
    domain: stateage.ages,
    scheme: "spectral",
    legend: true,
    marginLeft: 40
  },
```

**Comment by dgwyer** [comment]
Thanks, that works for the left margin but top, right, and bottom margins are ignored. I've updated the notebook to include these.

**Reply by mbostock** [reply]
The legend currently only supports the **marginLeft** option as documented here:

https://github.com/observablehq/plot#plotlegendscalename-options

**Reply by dgwyer** [reply]
Do you think categorical and ordinal color legends could be updated to support top, right, bottom margins, to match continuous color legends which look like they support all margins?

**Reply by dgwyer** [reply]
Would that be an easy update or are there more considerations?

**Reply by Fil** [reply]
It looks like it would be an easy update following https://github.com/observablehq/plot/blob/d4f0a8afbcff73d6655d2e0a72cd59e1a9fbda92/src/legends/swatches.js#L174, but I'm not sure about the use case.

**Reply by dgwyer** [reply]
I need to have a general margin for the legend, and not just the left. Seems a bit strange to not have it for categorical and ordinal color legends when it's already in place for continuous color legends. Would be good at least for consistency?

**Reply by dgwyer** [reply]
> It looks like it would be an easy update following

Thanks. I've just added a PR for full margin support. https://github.com/observablehq/plot/pull/1082
